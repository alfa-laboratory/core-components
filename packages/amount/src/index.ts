import { Amount as DefaultAmount } from './component';
import { PureAmount } from './pure/component';

type AmountType = typeof DefaultAmount & { Pure: typeof PureAmount };
export const Amount: AmountType = DefaultAmount as AmountType;
Amount.Pure = PureAmount;

export * from './types';
