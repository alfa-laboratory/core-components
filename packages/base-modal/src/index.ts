export * from './Component';
export * from './utils';
