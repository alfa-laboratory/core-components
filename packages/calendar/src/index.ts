export * from './Component';
export { usePeriod } from './usePeriod';
export { useCalendar } from './useCalendar';
export {
    russianWeekDay,
    dateInLimits,
    limitDate,
    monthName,
    DAYS_IN_WEEK,
    MONTHS,
    MONTHS_IN_YEAR,
    SUNDAY_INDEX,
    WEEKDAYS,
} from './utils';
