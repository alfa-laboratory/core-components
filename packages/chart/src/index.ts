export * from './Component';
export * from './types';
