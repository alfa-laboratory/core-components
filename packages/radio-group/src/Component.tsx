import React, {
    ReactNode,
    Children,
    cloneElement,
    ReactElement,
    ChangeEvent,
    MouseEvent,
    isValidElement,
    useState,
    forwardRef,
} from 'react';
import cn from 'classnames';

import styles from './index.module.css';

export type Direction = 'horizontal' | 'vertical';
export type RadioGroupType = 'radio' | 'tag';

export type RadioGroupProps = {
    /**
     * Заголовок группы
     */
    label?: ReactNode;

    /**
     * Направление
     */
    direction?: Direction;

    /**
     * Тип компонента
     */
    type?: RadioGroupType;

    /**
     * Дополнительный класс
     */
    className?: string;

    /**
     * Текст ошибки
     */
    error?: string;

    /**
     * Дочерние элементы. Ожидаются компоненты `Radio` или `Tag`
     */
    children: ReactNode;

    /**
     * Обработчик изменения значения 'checked' одного из дочерних компонентов
     */
    onChange?: (
        event?: ChangeEvent | MouseEvent,
        payload?: {
            value: string;
            name?: string;
        },
    ) => void;

    /**
     * Управление возможностью изменения состояния 'checked' дочерних компонентов Radio | Tag
     */
    disabled?: boolean;

    /**
     * Идентификатор для систем автоматизированного тестирования
     */
    dataTestId?: string;

    /**
     * Атрибут name для всех дочерних компонентов
     */
    name?: string;

    /**
     * Value выбранного дочернего элемента
     */
    value?: string;
};

export const RadioGroup = forwardRef<HTMLDivElement, RadioGroupProps>(
    (
        {
            children,
            className,
            direction = 'vertical',
            label,
            error,
            onChange,
            type = 'radio',
            dataTestId,
            disabled = false,
            name,
            value,
        },
        ref,
    ) => {
        const [stateValue, setStateValue] = useState<string>('');

        const renderRadio = (child: ReactElement) => {
            const { className: childClassName } = child.props;
            const checked = (value || stateValue) === child.props.value;
            const handleChange = (event: ChangeEvent) => {
                setStateValue(child.props.value);
                if (onChange) {
                    onChange(event, { name, value: child.props.value });
                }
            };

            return cloneElement(child, {
                onChange: handleChange,
                disabled,
                ...child.props,
                checked,
                name,
                className: cn(childClassName, styles.radio),
            });
        };

        const renderTag = (child: ReactElement) => {
            const checked = (value || stateValue) === child.props.value;
            const handleChange = (event: ChangeEvent | MouseEvent) => {
                setStateValue(child.props.value);
                if (onChange) {
                    onChange(event, { name, value: child.props.value });
                }
            };

            const clone = cloneElement(child, {
                onClick: handleChange,
                disabled,
                ...child.props,
                checked,
                name,
            });

            return (
                // eslint-disable-next-line jsx-a11y/label-has-associated-control
                <label className={cn(styles.radio, styles.tagLabel)}>
                    {clone}
                    <input
                        type='radio'
                        autoComplete='off'
                        onChange={handleChange}
                        disabled={disabled || child.props.disabled}
                        name={name}
                        checked={checked}
                        className={styles.hiddenInput}
                    />
                </label>
            );
        };

        return (
            <div
                className={cn(
                    styles.component,
                    styles[type],
                    styles[direction],
                    { [styles.error]: error },
                    className,
                )}
                data-test-id={dataTestId}
                ref={ref}
            >
                {label ? <span className={styles.label}>{label}</span> : null}

                {children ? (
                    <div className={cn(styles.radioList)}>
                        {Children.map(children, child => {
                            if (isValidElement(child)) {
                                return type === 'radio' ? renderRadio(child) : renderTag(child);
                            }

                            return null;
                        })}
                    </div>
                ) : null}

                {error && <span className={styles.errorMessage}>{error}</span>}
            </div>
        );
    },
);

/**
 * Для отображения в сторибуке
 */
RadioGroup.defaultProps = {
    direction: 'vertical',
    type: 'radio',
    disabled: false,
};
