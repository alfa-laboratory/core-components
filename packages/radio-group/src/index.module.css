@import '../../themes/src/default.css';

:root {
    --radio-group-error-color: var(--color-light-text-negative);
    --radio-group-label-color: var(--color-light-text-secondary);
}

.component {
    display: flex;
    flex-direction: column;
}

.error {
    padding-left: var(--gap-m);
    border-left: 1px solid var(--radio-group-error-color);
}

.radioList {
    display: flex;
}

.vertical .radioList {
    flex-direction: column;
    align-items: flex-start;
}

.vertical .radio {
    margin-bottom: var(--gap-m);

    &:last-child {
        margin-bottom: 0;
    }
}

.horizontal .radioList {
    flex-wrap: wrap;
    margin-right: var(--gap-xl-neg);
    margin-bottom: var(--gap-xs-neg);
}

.tag.horizontal .radioList {
    margin-right: var(--gap-xs-neg);
}

.horizontal .radio {
    margin-right: var(--gap-xl);
    margin-bottom: var(--gap-xs);
}

.horizontal .tagLabel {
    margin-right: var(--gap-xs);
}

.label {
    @mixin system_14-18_regular;
    margin-bottom: var(--gap-m);
    color: var(--radio-group-label-color);
}

.errorMessage {
    @mixin system_14-18_regular;
    margin-top: var(--gap-m);
    color: var(--radio-group-error-color);
}

.hiddenInput {
    position: absolute;
    z-index: -1;
    top: 0;
    left: 0;
    opacity: 0;
}

.tagLabel {
    position: relative;
}
