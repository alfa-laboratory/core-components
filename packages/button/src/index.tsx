import React, { forwardRef } from 'react';
import { Button as ButtonBase, ButtonProps } from './Component';

import styles from './index.module.css';

const Button = forwardRef<HTMLAnchorElement | HTMLButtonElement, ButtonProps>((props, ref) => {
    return <ButtonBase {...props} ref={ref} styles={styles} />;
});

export { Button, ButtonProps };
