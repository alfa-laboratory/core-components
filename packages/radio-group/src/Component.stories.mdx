import { Meta, Story, Props, Preview, Title } from '@storybook/addon-docs/blocks';
import { text, boolean, select } from '@storybook/addon-knobs';

import { RadioGroup } from './Component';
import { name, version } from '../package.json';
import { Radio } from '../../radio/src';
import { Tag } from '../../tag/src';

<Meta title='Компоненты|RadioGroup' component={RadioGroup} />

<Title>
    RadioGroup ({name}@{version})
</Title>

```tsx
import { RadioGroup } from '@alfalab/core-components-radio-group';
```

## Описание

Компонент группы радио.

<Story name='Песочница'>
    {React.createElement(() => {
        const [value, setValue] = React.useState({ one: false, two: false, three: false });
        const onChange = (_, payload) => {
            setValue({ ...value, [payload.name]: payload.checked });
        };
        const direction = select('direction', ['vertical', 'horizontal'], 'vertical');
        const error = text('error', '');
        const disabled = boolean('disabled', false);
        const label = text('label', 'Заголовок группы');
        return (
            <RadioGroup
                label={label}
                onChange={onChange}
                direction={direction}
                error={error}
                disabled={disabled}
                name='radioGroup'
            >
                <Radio label='Первый вариант' value='first' />
                <Radio label='Второй вариант' value='second' />
                <Radio label='Третий вариант' value='third' />
            </RadioGroup>
        );
    })}
</Story>

<Props of={RadioGroup} />

### Группа радио, использующая компонент `Tag`

<Preview>
    {React.createElement(() => {
        const [value, setValue] = React.useState({ one: false, two: false, three: false });
        const onChange = (_, payload) => {
            setValue({ ...value, [payload.name]: payload.checked });
        };
        return (
            <RadioGroup
                label="Заголовок группы"
                onChange={onChange}
                direction="horizontal"
                type='tag'
                name='tagRadioGroup'
            >
                <Tag value='one'>
                    Первый вариант
                </Tag>
                <Tag value='two'>
                    Второй вариант
                </Tag>
                <Tag value='three'>
                    Третий вариант
                </Tag>
            </RadioGroup>
        );
    })}
</Preview>
