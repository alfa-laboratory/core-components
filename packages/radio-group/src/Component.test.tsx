import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import { RadioGroup, RadioGroupProps } from './index';
import { Radio } from '../../radio/src';
import { Tag } from '../../tag/src';

const Group = ({ ...restProps }: Partial<RadioGroupProps>) => (
    <RadioGroup label='Заголовок группы' {...restProps}>
        <Radio label='Первый вариант' value='one' className='my-radio' />

        <Radio label='Второй вариант' value='two' className='my-second-radio' />

        <Radio label='Третий вариант' value='three' />
    </RadioGroup>
);

const TagGroup = ({ ...restProps }: Partial<RadioGroupProps>) => (
    <RadioGroup label='Заголовок группы' type='tag' {...restProps}>
        <Tag name='one' className='my-tag'>
            Первый вариант
        </Tag>

        <Tag name='two'>Второй вариант</Tag>

        <Tag name='three'>Третий вариант</Tag>
    </RadioGroup>
);

describe('RadioGroup', () => {
    describe('Display tests', () => {
        it('should display with children like boolean or string or others react children type correctly', () => {
            expect(
                render(
                    <RadioGroup label='Заголовок группы'>
                        {0 && <Radio label='Первый вариант' value='one' />}
                        {false && <Radio label='Первый вариант' value='one' />}
                        {null && <Radio label='Первый вариант' value='one' />}
                        {'' && <Radio label='Первый вариант' value='one' />}
                    </RadioGroup>,
                ),
            ).toMatchSnapshot();
        });

        it('should display radio group with one child correctly', () => {
            expect(
                render(
                    <RadioGroup label='Заголовок группы'>
                        <Radio label='Первый вариант' value='one' className='radio-group' />
                    </RadioGroup>,
                ),
            ).toMatchSnapshot();
        });

        it('should display radio group correctly', () => {
            expect(render(<Group />)).toMatchSnapshot();
        });

        it('should display tag radio group correctly', () => {
            expect(render(<TagGroup />)).toMatchSnapshot();
        });

        it('should display radio group with error correctly', () => {
            expect(render(<Group error='Error' />)).toMatchSnapshot();
        });
    });

    describe('Styles tests', () => {
        it('should set custom class', () => {
            const className = 'custom-class';
            const { container } = render(<Group className={className} />);

            expect(container.firstElementChild).toHaveClass(className);
        });

        it('should set `vertical` class', () => {
            const { container } = render(<Group direction='vertical' />);

            expect(container.firstElementChild).toHaveClass('vertical');
        });

        it('should set `error` class', () => {
            const { container } = render(<Group error='Error message' />);

            expect(container.firstElementChild).toHaveClass('error');
        });
    });

    describe('Attributes tests', () => {
        it('should set `data-test-id` atribute', () => {
            const dataTestId = 'test-id';
            const { container } = render(<Group dataTestId={dataTestId} />);

            const testIdAttr = container.firstElementChild?.getAttribute('data-test-id');

            expect(testIdAttr).toBe(dataTestId);
        });
    });

    describe('Render tests', () => {
        it('should unmount without errors', () => {
            const { unmount } = render(<Group />);

            expect(unmount).not.toThrowError();
        });

        it('should contain label', () => {
            const labelText = 'I am label';

            const { getByText } = render(<Group label={labelText} />);

            expect(getByText(labelText)).toBeInTheDocument();
        });

        it('should contain error message', () => {
            const errorText = 'I am error';

            const { getByText } = render(<Group error={errorText} />);

            expect(getByText(errorText)).toBeInTheDocument();
        });

        it('should disable all inputs if `disabled` prop is present', () => {
            const { container } = render(<Group disabled={true} />);

            const inputs = container.querySelectorAll('input');

            let allDisabled = true;

            inputs.forEach(input => {
                if (!input.disabled) {
                    allDisabled = false;
                }
            });

            expect(allDisabled).toBe(true);
        });
    });
});

describe('Interaction tests', () => {
    test('should call `onChange` callback if click on radio', () => {
        const cb = jest.fn();

        const { container } = render(<Group onChange={cb} />);

        const radio = container.querySelector('.my-radio');

        if (radio) {
            fireEvent.click(radio);
        }

        expect(cb).toBeCalledTimes(1);
    });

    test('should call `onChange` callback if click on tag', () => {
        const cb = jest.fn();

        const { container } = render(<TagGroup onChange={cb} />);

        const tag = container.querySelector('.my-tag');

        if (tag) {
            fireEvent.click(tag);
        }

        expect(cb).toBeCalledTimes(1);
    });

    test('should set "checked" classname for Radio with suitable value', () => {
        const { container } = render(<Group value='one' />);

        const radio = container.querySelector('.my-radio');

        expect(radio).toHaveClass('checked');
    });

    test('should pass "checked" classname to Radio after click', () => {
        const { container } = render(<Group />);

        const firstRadio = container.querySelector('.my-radio');
        const secondRadio = container.querySelector('.my-second-radio');
        expect(firstRadio).not.toHaveClass('checked');
        expect(secondRadio).not.toHaveClass('checked');
        if (firstRadio) {
            fireEvent.click(firstRadio);
        }
        expect(firstRadio).toHaveClass('checked');
        expect(secondRadio).not.toHaveClass('checked');
        if (secondRadio) {
            fireEvent.click(secondRadio);
        }
        expect(firstRadio).not.toHaveClass('checked');
        expect(secondRadio).toHaveClass('checked');
    });
});
