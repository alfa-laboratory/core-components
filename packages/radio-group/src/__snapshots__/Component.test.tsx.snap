// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup Display tests should display radio group correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="component radio vertical"
      >
        <span
          class="label"
        >
          Заголовок группы
        </span>
        <div
          class="radioList"
        >
          <label
            class="container my-radio radio"
          >
            <input
              type="radio"
              value="one"
            />
            <span
              class="circle"
            />
            <span
              class="content"
            >
              <span
                class="label"
              >
                Первый вариант
              </span>
            </span>
          </label>
          <label
            class="container my-second-radio radio"
          >
            <input
              type="radio"
              value="two"
            />
            <span
              class="circle"
            />
            <span
              class="content"
            >
              <span
                class="label"
              >
                Второй вариант
              </span>
            </span>
          </label>
          <label
            class="container radio"
          >
            <input
              type="radio"
              value="three"
            />
            <span
              class="circle"
            />
            <span
              class="content"
            >
              <span
                class="label"
              >
                Третий вариант
              </span>
            </span>
          </label>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="component radio vertical"
    >
      <span
        class="label"
      >
        Заголовок группы
      </span>
      <div
        class="radioList"
      >
        <label
          class="container my-radio radio"
        >
          <input
            type="radio"
            value="one"
          />
          <span
            class="circle"
          />
          <span
            class="content"
          >
            <span
              class="label"
            >
              Первый вариант
            </span>
          </span>
        </label>
        <label
          class="container my-second-radio radio"
        >
          <input
            type="radio"
            value="two"
          />
          <span
            class="circle"
          />
          <span
            class="content"
          >
            <span
              class="label"
            >
              Второй вариант
            </span>
          </span>
        </label>
        <label
          class="container radio"
        >
          <input
            type="radio"
            value="three"
          />
          <span
            class="circle"
          />
          <span
            class="content"
          >
            <span
              class="label"
            >
              Третий вариант
            </span>
          </span>
        </label>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`RadioGroup Display tests should display radio group with error correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="component radio vertical error"
      >
        <span
          class="label"
        >
          Заголовок группы
        </span>
        <div
          class="radioList"
        >
          <label
            class="container my-radio radio"
          >
            <input
              type="radio"
              value="one"
            />
            <span
              class="circle"
            />
            <span
              class="content"
            >
              <span
                class="label"
              >
                Первый вариант
              </span>
            </span>
          </label>
          <label
            class="container my-second-radio radio"
          >
            <input
              type="radio"
              value="two"
            />
            <span
              class="circle"
            />
            <span
              class="content"
            >
              <span
                class="label"
              >
                Второй вариант
              </span>
            </span>
          </label>
          <label
            class="container radio"
          >
            <input
              type="radio"
              value="three"
            />
            <span
              class="circle"
            />
            <span
              class="content"
            >
              <span
                class="label"
              >
                Третий вариант
              </span>
            </span>
          </label>
        </div>
        <span
          class="errorMessage"
        >
          Error
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="component radio vertical error"
    >
      <span
        class="label"
      >
        Заголовок группы
      </span>
      <div
        class="radioList"
      >
        <label
          class="container my-radio radio"
        >
          <input
            type="radio"
            value="one"
          />
          <span
            class="circle"
          />
          <span
            class="content"
          >
            <span
              class="label"
            >
              Первый вариант
            </span>
          </span>
        </label>
        <label
          class="container my-second-radio radio"
        >
          <input
            type="radio"
            value="two"
          />
          <span
            class="circle"
          />
          <span
            class="content"
          >
            <span
              class="label"
            >
              Второй вариант
            </span>
          </span>
        </label>
        <label
          class="container radio"
        >
          <input
            type="radio"
            value="three"
          />
          <span
            class="circle"
          />
          <span
            class="content"
          >
            <span
              class="label"
            >
              Третий вариант
            </span>
          </span>
        </label>
      </div>
      <span
        class="errorMessage"
      >
        Error
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`RadioGroup Display tests should display radio group with one child correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="component radio vertical"
      >
        <span
          class="label"
        >
          Заголовок группы
        </span>
        <div
          class="radioList"
        >
          <label
            class="container radio-group radio"
          >
            <input
              type="radio"
              value="one"
            />
            <span
              class="circle"
            />
            <span
              class="content"
            >
              <span
                class="label"
              >
                Первый вариант
              </span>
            </span>
          </label>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="component radio vertical"
    >
      <span
        class="label"
      >
        Заголовок группы
      </span>
      <div
        class="radioList"
      >
        <label
          class="container radio-group radio"
        >
          <input
            type="radio"
            value="one"
          />
          <span
            class="circle"
          />
          <span
            class="content"
          >
            <span
              class="label"
            >
              Первый вариант
            </span>
          </span>
        </label>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`RadioGroup Display tests should display tag radio group correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="component tag vertical"
      >
        <span
          class="label"
        >
          Заголовок группы
        </span>
        <div
          class="radioList"
        >
          <label
            class="radio tagLabel"
          >
            <button
              class="component s my-tag"
              type="button"
            >
              <span>
                Первый вариант
              </span>
            </button>
            <input
              autocomplete="off"
              class="hiddenInput"
              type="radio"
            />
          </label>
          <label
            class="radio tagLabel"
          >
            <button
              class="component s"
              type="button"
            >
              <span>
                Второй вариант
              </span>
            </button>
            <input
              autocomplete="off"
              class="hiddenInput"
              type="radio"
            />
          </label>
          <label
            class="radio tagLabel"
          >
            <button
              class="component s"
              type="button"
            >
              <span>
                Третий вариант
              </span>
            </button>
            <input
              autocomplete="off"
              class="hiddenInput"
              type="radio"
            />
          </label>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="component tag vertical"
    >
      <span
        class="label"
      >
        Заголовок группы
      </span>
      <div
        class="radioList"
      >
        <label
          class="radio tagLabel"
        >
          <button
            class="component s my-tag"
            type="button"
          >
            <span>
              Первый вариант
            </span>
          </button>
          <input
            autocomplete="off"
            class="hiddenInput"
            type="radio"
          />
        </label>
        <label
          class="radio tagLabel"
        >
          <button
            class="component s"
            type="button"
          >
            <span>
              Второй вариант
            </span>
          </button>
          <input
            autocomplete="off"
            class="hiddenInput"
            type="radio"
          />
        </label>
        <label
          class="radio tagLabel"
        >
          <button
            class="component s"
            type="button"
          >
            <span>
              Третий вариант
            </span>
          </button>
          <input
            autocomplete="off"
            class="hiddenInput"
            type="radio"
          />
        </label>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`RadioGroup Display tests should display with children like boolean or string or others react children type correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="component radio vertical"
      >
        <span
          class="label"
        >
          Заголовок группы
        </span>
        <div
          class="radioList"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="component radio vertical"
    >
      <span
        class="label"
      >
        Заголовок группы
      </span>
      <div
        class="radioList"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
